app/Http/Controllers/TaskController.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Support\Facades\Validator;
+
+use Illuminate\Validation\Rule;
+
+use Illuminate\Http\Request;
+
+Use App\Task;
+
+class TaskController extends Controller
+{
+
+    public function create($search_id)
+    {
+        if (\Auth::check()) {
+            $currentUser = \Auth::user()->profile;
+
+            if ($currentUser != 'guest') {
+                return view('searches.tasks.create', [
+                    'search_id' => $search_id
+                ]);
+            } else {
+                return redirect('searches/'.$incident->search_id)
+                ->with('error', __('messages.not_allowed'));
+            }
+        } else {
+            return redirect()->action('HomeController@login');
+        }
+    }
+
+    public function store(Request $request)
+    {
+    	$request->validate([
+            'search_id' => 'required',
+            'Sector' => 'required',
+            'Status' => 'required',Rule::in(["to_do","in_progress","done"]),
+            'Group' => 'required',
+            'Start' => 'required|date',
+            'End' => 'required|date|after:Start', 
```

Comment:
The after:Start condition is okey but this validation should be on the frontend. I have an example on the search form where the calendar component disable past dates. The frontend validation for dates is much better for users